src/plots/cartesian/axes.js

```diff
@@ -455,6 +455,13 @@ axes.expand = function(ax, data, options) {
         i, j, v, di, dmin, dmax,
         ppadiplus, ppadiminus, includeThis, vmin, vmax;
 
+    // domain-constrained axes: base extrappad on the unconstrained
+    // domain so it's consistent as the domain changes
+    if(extrappad && ax._inputDomain) {
```

Comment:
Interesting choice for having a condition based on `ax._inputDomain`. 

Wouldn't `ax.constrain === 'domain'` be more robust?